spec/unit/node_spec.rb

```diff
@@ -219,6 +219,12 @@ def from_json(json)
     expect(@node.parameters["two"]).to eq("b")
   end
 
+  it "warns when a parameter value is not updated" do
+    @node = Puppet::Node.new("testnode", :parameters => {"one" => "a"})
+    Puppet.expects(:warning).with('The node parameter one for node  was already set to a. It could not be set to b')
```

Comment:
there is no node name in the message, and the name of the parameter should be in single quotes (as commented on above).